service/src/main/java/uk/nhs/adaptors/gp2gp/common/storage/S3StorageConnector.java

```diff
@@ -2,44 +2,41 @@
 
 import java.io.InputStream;
 
-import com.amazonaws.services.s3.AmazonS3;
-import com.amazonaws.services.s3.AmazonS3ClientBuilder;
-import com.amazonaws.services.s3.model.ObjectMetadata;
-import com.amazonaws.services.s3.model.S3Object;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
 
 public class S3StorageConnector implements StorageConnector {
-    private final AmazonS3 s3client;
+    private final S3Client s3client;
     private final String bucketName;
 
-    protected S3StorageConnector(StorageConnectorConfiguration configuration) {
+    protected S3StorageConnector(S3Client s3client, StorageConnectorConfiguration configuration) {
         this.bucketName = configuration.getContainerName();
-        this.s3client = AmazonS3ClientBuilder
-            .standard()
-            .build();
+        this.s3client = s3client;
```

Comment:
Will this use the Bean defined as `getS3Client` inside of `StorageConnectorConfiguration.java` ? Because if so, then that bean only gets constructed when a truststore is defined. Whereas previously this was being constructed all the time.